cpp/tests/sort/rank_test.cpp

```diff
@@ -444,3 +452,487 @@ TEST_F(RankLarge, average_large)
   cudf::test::fixed_width_column_wrapper<double, int> expected(iter + 1, iter + 10559);
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
 }
+
+template <typename T>
+struct RankListAndStruct : public cudf::test::BaseFixture {
+  void run_all_tests(cudf::rank_method method,
+                     input_arg_t input_arg,
+                     cudf::column_view const list_rank,
+                     cudf::column_view const struct_rank,
+                     bool percentage = false)
+  {
+    if constexpr (std::is_same_v<T, bool>) { return; }
+    /*
+    [
+      [],
+      [1],
+      [2, 2],
+      [2, 3],
+      [2, 2],
+      [1],
+      [],
+      NULL
+      [2],
+      NULL,
+      [1]
+    ]
+    */
+    auto list_col =
+      lists_col<T>{{{}, {1}, {2, 2}, {2, 3}, {2, 2}, {1}, {}, {} /*NULL*/, {2}, {} /*NULL*/, {1}},
+                   nulls_at({7, 9})};
+
+    // clang-format off
+    /*
+      +------------+
+      |           s|
+      +------------+
+    0 |   {0, null}|
+    1 |   {1, null}|
+    2 |        null|
+    3 |{null, null}|
+    4 |        null|
+    5 |{null, null}|
+    6 |   {null, 1}|
+    7 |   {null, 0}|
+      +------------+
+    */
+    std::vector<bool>                           struct_valids{1, 1, 0, 1, 0, 1, 1, 1};
+    auto col1       = cudf::test::fixed_width_column_wrapper<T>{{ 0,  1,  9, -1,  9, -1, -1, -1}, {1, 1, 1, 0, 1, 0, 0, 0}};
+    auto col2       = cudf::test::fixed_width_column_wrapper<T>{{-1, -1,  9, -1,  9, -1,  1,  0}, {0, 0, 1, 0, 1, 0, 1, 1}};
+    auto struct_col = cudf::test::structs_column_wrapper{{col1, col2}, struct_valids}.release();
+    //clang-format on
```

Comment:
I think a space may be needed here to re-enable clang-format correctly.

I usually prefer to let the formatter do its job and avoid manual alignment of test data, but I don't have strong feelings.
```suggestion
    // clang-format on
```